src/main/java/software/amazon/awssdk/crt/http/Http2StreamManagerOptions.java

```diff
@@ -0,0 +1,159 @@
+        return idealConcurrentStreamsPerConnection;
+    }
+
+    /**
+     * Default is no limit, which will use the limit from the server. 0 will be
+     * considered as using the default value.
+     * The real number of concurrent streams per connection will be controlled by
+     * the minimal value of the setting from other end and the value here.
+     *
+     * @param maxConcurrentStreamsPerConnection The max number of concurrent
+     *                                          streams for a connection
+     * @return this
+     */
+    public Http2StreamManagerOptions withMaxConcurrentStreamsPerConnection(int maxConcurrentStreamsPerConnection) {
+        this.maxConcurrentStreamsPerConnection = maxConcurrentStreamsPerConnection;
+        return this;
+    }
+
+    /**
+     * @return The max number of concurrent streams for a connection set for
+     *         manager.
+     *         It could be different than the real limits, which is the minimal set
+     *         for manager and the settings from the other side.
+     */
+    public int getMaxConcurrentStreamsPerConnection() {
+        return maxConcurrentStreamsPerConnection;
+    }
+
+    /**
+     * @return The connection level manual flow control enabled or not.
+     */
+    public boolean isConnectionManualWindowManagement() {
+        return connectionManualWindowManagement;
+    }
+
+    /**
+     * Set to true to manually manage the flow-control window of whole HTTP/2 connection.
+     * The stream level flow-control window is controlled by the manualWindowManagement in connectionManagerOptions.
+     *
+     * @param connectionManualWindowManagement Enable connection level manual flow control or not.
+     * @return this
+     */
+    public Http2StreamManagerOptions withConnectionManualWindowManagement(boolean connectionManualWindowManagement) {
+        this.connectionManualWindowManagement = connectionManualWindowManagement;
+        return this;
+    }
+
+    /**
+     * @return The connection manager options for the underlying connection manager.
+     */
+    public HttpClientConnectionManagerOptions getConnectionManagerOptions() {
+        return connectionManagerOptions;
+    }
+
+    /**
+     * The configuration options for the connection manager under the hood.
+     * It controls the connection specific thing for the stream manager. See `HttpClientConnectionManagerOptions` for details.
+     *
+     * Note:
+     *    1. the windowSize of connection manager will be ignored, as the initial flow-control window size for HTTP/2 stream
+     *       is controlled by the initial settings.
+     *    2. The expectedHttpVersion will also be ignored.
+     *
+     * @param connectionManagerOptions The connection manager options for the underlying connection manager
+     * @return this
+     */
+    public Http2StreamManagerOptions withConnectionManagerOptions(HttpClientConnectionManagerOptions connectionManagerOptions) {
```

Comment:
Should we denote that this is required? (how do we usually indicate that something is required in an options class?)